crates/task-impls/src/da.rs

```diff
@@ -247,23 +164,18 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     .saved_payloads
                     .insert(payload_commitment, proposal.data.encoded_transactions);
             }
-            HotShotEvent::DAVoteRecv(vote) => {
-                debug!("DA vote recv, Main Task {:?}", vote.get_view_number(),);
+            HotShotEvent::DAVoteRecv(ref vote) => {
+                debug!("DA vote recv, Main Task {:?}", vote.get_view_number());
                 // Check if we are the leader and the vote is from the sender.
                 let view = vote.get_view_number();
                 if self.da_membership.get_leader(view) != self.public_key {
                     error!("We are not the committee leader for view {} are we leader for next view? {}", *view, self.da_membership.get_leader(view + 1) == self.public_key);
                     return None;
                 }
```

Comment:
If we are going to remove the leadership check from each task (not sure if we should though, see my other comment), should we also move it here?